io/net/virt-net/Network_virtualization_dlpar.py

```diff
@@ -160,34 +170,49 @@ def test(self):
             self.fail("Failed to start the rsct and rsct_rm services")
 
         for _ in range(self.count):
-            self.network_virtualization__add()
+            self.network_virtualization_add()
+            self.network_virtualization_backing_device_add()
+            self.network_virtualization_backing_device_remove()
```

Comment:
Ideally, these 4 should be 4 seperate `test_x()` functions

Reply:
When I prefix 'test' to it., I get 5 variants of tests being run. but they run based on the order in which they occur in the script. 


[root@ltc-zz3-lp15 TEST]# avocado run Network_virtualization.py -m Network_virtualization.py.data/Network_virtualization.yaml 
JOB ID     : 0f06e8a8486ad4b1036152f04085ede68707384d
JOB LOG    : /root/avocado/job-results/job-2018-03-30T08.54-0f06e8a/job.log
 (1/5) Network_virtualization.py:NetworkVirtualization.test;run-62e5: FAIL (45.74 s)
 (2/5) Network_virtualization.py:NetworkVirtualization.test_network_virtualization_remove;run-62e5: PASS (79.62 s)
 (3/5) Network_virtualization.py:NetworkVirtualization.test_network_virtualization_add;run-62e5: PASS (56.94 s)
 (4/5) Network_virtualization.py:NetworkVirtualization.test_network_virtualization_backing_device_remove;run-62e5: FAIL (37.33 s)
 (5/5) Network_virtualization.py:NetworkVirtualization.test_network_virtualization_backing_device_add;run-62e5: PASS (53.82 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 273.66 s


This is not the right order. Need suggestion on what can be done to corrcet the order of the script run

Reply:
I don't think `test()` is required anymore.
You can re-order the functions. You are free to choose the order logically.
Example:
```
test_network_virtualization_add
test_network_virtualization_remove
test_network_virtualization_backing_device_add
test_network_virtualization_backing_device_remove
```

Reply:
Also, the existing `service start` part can be transformed into a function and called from `setUp()` instead.

Reply:
Ok. I will implement this. 

Right now, Stuck at something really simple, (I guess)

    for _ in range(self.count):
        self.test_network_virtualization_add()
        #self.test_network_virtualization_backing_device_add()
        #self.test_network_virtualization_backing_device_remove()
        #self.test_network_virtualization_remove()


I see the below error:
    for _ in range(self.count):
NameError: name 'self' is not defined


Even when I have defined count as this:
self.count = int(self.params.get("count", '*', default="1"))

Reply:
Sure. Will include these changes.

Reply:
@narasimhan-v After including the yaml parameter, 
iteration: !mux
    1:
    2:

I found the tests to run in this order

test1
test1

test2
test2

===> Finding out how I can run it in order
test1
test2

test1
test2

Reply:
Used --execution-order tests-per-variant in the command line to get this working. 
This has been  documented in README